interfaces/automated-testing/scd/design/strategic-coordination-interoperability/nominal-test.plantuml

```diff
@@ -0,0 +1,28 @@
+@startuml
+center header Strategic Coordination Interoperability
+title Nominal test
+
+participant "Test Driver" as TD
+participant "USS1 under test" as USS1
+participant "USS2 under test" as USS2
+participant "DSS"
+
+== TD injects via USS1 non-conflicting Operational Intent 1 ==
+
+TD --> USS1: PUT /inject/operational_intent
+TD <-- USS1: 200 OK - Operational Intent 1 created
+
+TD --> DSS: POST /dss/v1/operational_intent_references/query\nwith area_of_interest encompassing Operational Intent 1
```

Comment:
This is more of a question for both of you, do we really want the test driver to query the DSS (I don't have a problem), see the other comment, having a mock USS / reference implementation could be a better approach. I don't have a strong preference, but my gut feeling is that the test driver could be as simple as possible.

Reply:
In an ideal world, USSs would implement each of the appropriate standards and nothing else, then we would write an automated testing tool which would test that system.  The standards don't yet contain enough specification to accomplish this though, so we are forced to create some additional interfaces we need USSs to implement in order to be tested by our tools (hopefully these interfaces or something like them will be incorporated in later versions of the standards).  We should write these interfaces to be as simple as possible for USSs to implement while still allowing us to test the things we need to test.  In the case of RID, there is no interface defined in the standard that we could use to determine which flights a Display Provider sees, so we needed an observation interface to accomplish that goal.  For SCD, there is an interface defined in the standard that allows us to determine what flights (operational intents) exist in the system (query DSS for references, then query USSs for details).  We should definitely use this pre-existing interface to determine what operational intents are present in the system.  So, after we instruct a USS to create a flight that we expect to be created successfully, we can then definitively verify whether that operational intent was emplaced in the system by querying the DSS (and then following up for details with the USS).

TL;DR: Yes, we should definitely query the DSS to retrieve ground truth.  The only reason it was different for RID was because an equivalent interface didn't exist there.

Reply:
Thanks yes, see above comments, I will start making the changes.